internal/testutils/integrationHelpers.go

```diff
@@ -66,8 +66,19 @@ func ConnectAndMigrate(t *testing.T) *gorm.DB {
 func Truncate(db *gorm.DB) error {
 	// gorm doesn't seem to support truncate operations which are essential to cleaning up after
 	// integration tests (and the only use case of this function so doing it with raw sql)
-	truncateStatement := "TRUNCATE TABLE services, builds, environments, service_instances, deploys, clusters, allocation_pools"
-	err := db.Exec(truncateStatement).Error
+	testDeleteStatement := `
```

Comment:
What's the reason to move to DELETE from TRUNCATE? I honestly have no idea what the difference is so just curious.

Reply:
The truncates were the cause of the race condition I was mentioning. Postgres truncates on tables with very few rows can be very slow. They were actually causing deadlocks when there were many calls to the truncate function happening in quick succession.

Reply:
:mindblown: nice find!

Reply:
er also, I'm cool w/ just nuking the old, commented truncate statement in that case.

Reply:
Thanks! Yeah if we were seeding and then dropping very large tables in tests then I think truncate would be the right tool for that job.